client-config/src/main/java/com/palantir/conjure/java/client/config/ClientConfigurations.java

```diff
@@ -202,6 +207,44 @@ public static ProxySelector createProxySelector(ProxyConfiguration proxyConfig)
                 UnsafeArg.of("hostAndPort", proxyConfig.hostAndPort()));
     }
 
+    @VisibleForTesting
+    static ProxySelector proxySelectorFromEnvironment(@Nullable String defaultEnvProxy) {
+        if (defaultEnvProxy == null) {
+            log.info("Proxy environment variable not set, using no proxy");
+            return fixedProxySelectorFor(Proxy.NO_PROXY);
+        }
+        log.info("Using proxy from environment variable", UnsafeArg.of("proxy", defaultEnvProxy));
+        InetSocketAddress address = createInetSocketAddress(defaultEnvProxy);
+        return fixedProxySelectorFor(new Proxy(Proxy.Type.HTTP, address));
+    }
+
+    @VisibleForTesting
+    static ProxySelector proxySelectorFromEnvironmentNoProxyAware(
```

Comment:
It's tricky to set the OS level environment variables in a unit test. When I tried using JUnit Pioneer (which has an annotation for setting environment variables in JUnit tests), I ran into an error pointing to [this page](https://junit-pioneer.org/docs/environment-variables/#warnings-for-reflective-access) which says:

> The best way to prevent these warnings/errors, is to change the code under test, so this extension is no longer needed

I'm opting to use these helper methods instead.

Happy to split out this PR so that I refactor existing `FROM_ENVIRONMENT` logic first.